contracts/FeeDisbursal/FeeDisbursal.sol

```diff
@@ -0,0 +1,269 @@
+
+    // map token addresses to a list of disbursal events of that token
+    mapping (address => Disbursal[]) private _disbursals;
+    // mapping to track what disbursals have already been paid out to which user
+    // token address => user address => disbursal progress
+    mapping (address => mapping(address => uint256)) _disbursalProgress;
+
+
+    ////////////////////////
+    // Constructor
+    ////////////////////////
+    constructor(Universe universe)
+        AccessControlled(universe.accessPolicy())
+        Reclaimable()
+        public
+    {
+        UNIVERSE = universe;
+        IDENTITY_REGISTRY = IIdentityRegistry(universe.identityRegistry());
+    }
+
+    ////////////////////////
+    // Public functions
+    ////////////////////////
+
+    /// @notice get the disbursal at a given index for a given token
+    /// @param token address of the claimable token
+    /// @param index until what index to claim to
+    function getDisbursal(address token, uint256 index)
+    public
+    view
+    returns (
+        uint256 snapshotId,
+        uint256 amount,
+        BasicSnapshotToken proRataToken,
+        uint256 recycleAfterTimestamp,
+        address disburser
+    )
+    {
+        Disbursal storage disbursal = _disbursals[token][index];
+        snapshotId = disbursal.snapshotId;
+        amount = disbursal.amount;
+        proRataToken = disbursal.proRataToken;
+        recycleAfterTimestamp = disbursal.recycleAfterTimestamp;
+        disburser = disbursal.disburser;
+    }
+
+    /// @notice claim a token, to be called an investor
+    /// @param token address of the claimable token
+    /// @param until until what index to claim to
+    function claim(address token, uint256 until)
+    public
+    returns (uint256 claimableAmount, uint256 lastIndex)
+    {
+        // only allow verified and active accounts to claim tokens
+        // @TODO: move access control to Controller
+        IdentityClaims memory claims = deserializeClaims(IDENTITY_REGISTRY.getClaims(msg.sender));
+        require(claims.isVerified && !claims.accountFrozen, "NF_DISB_NOT_VER");
+        return claimPrivate(token, msg.sender, until);
+    }
+
+    /// @notice claim multiple tokens, to be called an investor
+    /// @param tokens addresses of the claimable token
+    function claimMultiple(address[] tokens)
+    public
+    returns (uint256[])
+    {
+        // only allow verified and active accounts to claim tokens
+        // @TODO: move access control to Controller
+        IdentityClaims memory claims = deserializeClaims(IDENTITY_REGISTRY.getClaims(msg.sender));
+        require(claims.isVerified && !claims.accountFrozen, "NF_DISB_NOT_VER");
+        uint256[] memory result = new uint256[](tokens.length);
+        for (uint256 i = 0; i < tokens.length; i += 1) {
+            (uint256 claimedAmount, uint256 lastIndex)  = claimPrivate(tokens[i], msg.sender, UINT256_MAX);
+            result[i] = claimedAmount;
+        }
+        return result; 
```

Comment:
please note that external clients do not have access to return value of non-constant function. this is only available for intra contract calls.
also maybe this function should be `external`
or maybe you plan to call it with `eth_call` as a kind of dry run? then it makes sense

Reply:
No I was not planning that. Thanks for pointing this out, I will remove all the returns for non constant functions